Flow IsTrimmable from FrameworkReference to resolved assets

sbomer opened the pull request:
Follow-up to https://github.com/dotnet/core-sdk/pull/1992/files. The next step is to flow the metadata from `RuntimePack` to the resolved files.

This flows `IsTrimmable` metadata from `FrameworkReference` and `KnownFrameworkReference` (`FrameworkReference` has priority) to the `RuntimePack` ItemGroup produced by `ResolveFrameworkReferences`. The metadata will ultimately influence the default linker behavior.

@dsplaisted @nguerrera PTAL.

sbomer committed:
Flow IsTrimmable from FrameworkReference to RuntimePack

dsplaisted commented:
Don't you want to also flow the IsTrimmable metadata to the DLLs that come out of ResolveRuntimePackAssets?

sbomer committed:
Flow IsTrimmable to runtime pack assets

The test now checks that this metadata exists on ResolvedFileToPublish
items that come from a frameworkreference with IsTrimmable set.

sbomer commented:
Was going to do it in a separate PR, but looks like it will pretty much be a one-line addition, though it changes testing a little bit. I'm going to make the update once https://github.com/dotnet/core-sdk/pull/1992 is in the stage0, so that I can test against the real KnownFrameworkReference instead of a mocked one (since I need to actually restore the framework to have the test check metadata on the resolved assets).

sbomer committed:
Update stage0 to get IsTrimmable metadata

From https://github.com/dotnet/core-sdk/pull/1992

sbomer commented:
@dsplaisted @nguerrera PTAL

nguerrera commented on src/Tests/Microsoft.NET.Build.Tests/GivenFrameworkReferences.cs:
```diff
@@ -640,6 +685,78 @@ private List<string> GetRuntimeFrameworks(string runtimeConfigPath)
+    <ItemGroup>
+        <LinesToWrite Include=`%(ResolvedFileToPublish.Identity)%09%(ResolvedFileToPublish.PackageName)%09%(ResolvedFileToPublish.IsTrimmable)` />
+    </ItemGroup>
+    <WriteLinesToFile File=`$(OutputPath)runtimeassettriminfo.txt`
```
You should be able to use the existing GetValuesCommand for this.

sbomer replied:
Oh that looks much better, thanks!

sbomer replied:
I've addressed your feedback. Would you mind taking another look?

sbomer committed:
Update test to use GetValuesCommand
